File: vacancy-app/src/main/java/a/gleb/vacancy_app/db/entity/BaseEntity.java
package a.gleb.vacancy_app.db.entity;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.experimental.SuperBuilder;
import org.hibernate.annotations.UuidGenerator;

import java.time.LocalDateTime;
import java.util.UUID;

@Getter
@SuperBuilder
@MappedSuperclass
@NoArgsConstructor
public abstract class BaseEntity {

    @Id
    @UuidGenerator(style = UuidGenerator.Style.VERSION_7)
    private UUID id;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at")
    private LocalDateTime updatedAt;

    @Column(name = "created_by")
    private UUID createdBy;

    @Column(name = "updated_by")
    private UUID updatedBy;

    public void onCreate() {
        this.createdAt = LocalDateTime.now();
    }

    public void onUpdate() {
        this.updatedAt = LocalDateTime.now();
    }
}

File: main-service/src/main/kotlin/a/gleb/main_service/db/entity/BaseEntity.kt
package a.gleb.main_service.db.entity

import org.springframework.data.annotation.Id
import org.springframework.data.relational.core.mapping.Column
import java.time.LocalDateTime
import java.util.*

abstract class BaseEntity(
    @Id
    var id: UUID? = null,
    @Column(value = "created_at")
    var createdAt: LocalDateTime = LocalDateTime.now(),
    @Column(value = "updated_at")
    var updatedAt: LocalDateTime = LocalDateTime.now()
) {
}